File: backend/app/models/github.py
from pydantic import BaseModel
from typing import List, Optional

class TokenValidationResponse(BaseModel):
    valid: bool
    username: Optional[str] = None

class RepositoryResponse(BaseModel):
    id: int
    name: str
    full_name: str
    description: Optional[str] = None
    archived: bool
    updated_at: str
    stars: int
    language: Optional[str] = None

class RepositoryList(BaseModel):
    repositories: List[RepositoryResponse]
    total_count: int
    current_page: int
    total_pages: int

class ArchiveRequest(BaseModel):
    repository_ids: List[int]

class FailedRepository(BaseModel):
    id: int
    name: str
    error: str

class ArchiveResponse(BaseModel):
    success: bool
    archived_count: int
    failed_repositories: List[FailedRepository] = [] 

File: frontend/src/types/github.ts
export interface Repository {
    id: number;
    name: string;
    full_name: string;
    description?: string;
    archived: boolean;
    updated_at: string;
    stars: number;
    language?: string;
}

export interface RepositoryList {
    repositories: Repository[];
    total_count: number;
    current_page: number;
    total_pages: number;
}

export interface TokenValidation {
    valid: boolean;
    username?: string;
}

export interface ArchiveRequest {
    repository_ids: number[];
}

export interface FailedRepository {
    id: number;
    name: string;
    error: string;
}

export interface ArchiveResponse {
    success: boolean;
    archived_count: number;
    failed_repositories: FailedRepository[];
} 